ElectronClient/app/gui/NotebookPropertiesDialog.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+
+const { _ } = require('lib/locale.js');
+const { themeStyle } = require('../theme.js');
+const DialogButtonRow = require('./DialogButtonRow.min');
+
+const Folder = require('lib/models/Folder.js');
+
+interface NotebookPropertiesDialogProps {
+	theme: number,
+	folderId: string,
+	onClose: Function,
+}
+
+export default function NotebookPropertiesDialog(props: NotebookPropertiesDialogProps) {
+	console.info('Render NotebookPropertiesDialog');
+
+	const [notebook, setNotebook] = useState<any>({});
+	const [icon, setIcon] = useState<string>('fa fa-book'); //
+	const [loaded, setLoaded] = useState<boolean>(false);
+
+	const theme = themeStyle(props.theme);
+
+	useEffect(() => {
+		async function fetchNotebook() {
+			if (!props.folderId) {
+				setNotebook({notebook: null});
+			} else {
+				const n = await Folder.load(props.folderId);
+				const nb = formNotebook(n);
+				setNotebook(nb);
+			}
+		}
+
+		fetchNotebook();
+	}, [props.folderId]);
+
+	const formNotebook = (targetNotebook:any) => {
+		if (!loaded) {
+			const nb = Object.assign({}, targetNotebook, {
+				icon: targetNotebook.icon,
+			});
+			setIcon(nb.icon);
+			setLoaded(true);
```

Comment:
If you have a `formFolder` state variable, you don't need to also have an icon state variable, because there's already `formFolder.icon`.

Reply:
In my current test branch, this has been removed for that reason.